Use Travis CI to run tests on all code changes

cclauss opened the pull request:
Every time someone submits a pull request, Travis Continuous Integration service can run the tests on it before it is reviewed.  Travis is always free to opensource projects like this one but the owner of the this repo would need to go to https://travis-ci.org/profile and flip the repository switch on to enable testing.

Commit:
Use Travis CI to run tests on all code changes

Every time someone submits a pull request, Travis Continuous Integration service can run the tests on it before it is reviewed.  Travis is always free to opensource projects like this one but the owner of the this repo would need to go to https://travis-ci.org/profile and flip the repository switch on to enable testing.

lorenzb commented:
Thanks for your PR @cclauss!

Travis jobs are [automatically killed after 50 minutes](https://docs.travis-ci.com/user/customizing-the-build#Build-Timeouts) and our full test suite typically takes longer than that.

A potential solution might be to split the travis build into a few different jobs.

Commit:
Disable the tests for now.

Understood.  If you turn the Travis CI switch __on__ then I can tune the tests runs base on actual timings.

lorenzb commented:
Travis is enabled

lorenzb commented on .travis.yml:
```diff
@@ -0,0 +1,29 @@
+language: python
+cache: pip
+python:
+    - 2.7
```
We only use Python >=3.6

Commit:
Python 3.6 only... Let's time each test

Commit:
Remove all mention on Python 2.  Also remove allow_failures

I will need your guidance on why the __tests are not running__:
* https://travis-ci.org/projectchicago/gastoken/builds/351670243#L617-L769

cclauss commented:
I will need your guidance on why the __tests are not running__:
* https://travis-ci.org/projectchicago/gastoken/builds/351671585#L576-L728

cclauss commented:
This one sets a new record!  https://travis-ci.org/projectchicago/gastoken/builds/351671585#L499

;-)

lorenzb commented:
Tests are failing because the solidity compiler `solc` isn't available in the travis environment.
Installation instructions are [here](https://solidity.readthedocs.io/en/v0.4.20/installing-solidity.html).

Commit:
npm install -g solc

Commit:
snap install solc

Commit:
udo apt update ; sudo apt install snapd ; sudo snap install solc

cclauss commented:
['elopio' installed](https://travis-ci.org/projectchicago/gastoken/builds/351692281#L726) but it does not seem to help.

cclauss commented:
@lorenzb When you get a minute, can you take a look at https://travis-ci.org/projectchicago/gastoken/builds/351692281 and let me know what is missing?
